src/subdomains/staking/application/factories/staking.factory.ts

```diff
@@ -60,8 +64,28 @@ export class StakingFactory {
       outputReferenceAmount,
       feePercent,
       feeAmount,
-      targetAsset,
-      targetAddress,
+      rewardRoute,
     );
   }
+
+  createRewardRoute(staking: Staking, dto: CreateRewardRouteDto, supportedAssets: Asset[]): RewardRoute {
+    const { label, rewardPercent, targetAssetName, targetAddress: targetAddressName, targetBlockchain } = dto;
+
+    const targetAsset =
+      targetAssetName === 'DFI'
+        ? supportedAssets.find(
+            (a) => a.name === targetAssetName && a.blockchain === targetBlockchain && a.type === AssetType.COIN,
+          )
+        : supportedAssets.find((a) => a.name === targetAssetName && a.blockchain === targetBlockchain);
```

Comment:
This will probably be a general pattern. If there is a coin for the selected asset, take it, otherwise take the token.